lib/index.js

```diff
@@ -149,36 +195,42 @@ export default {
           linterName => configEnabledEngines.has(linterName));
 
         // Construct the command line invocation which runs the Code Climate CLI
-        const cmd = [this.executablePath, 'analyze',
-          '-f json',
-          makeEngineString(linterEnabledEngines),
-          `'${atom.project.relativize(filePath)}'`,
-          '< /dev/null'].join(' ');
+        const relpath = atom.project.relativizePath(filePath).pop();
+        const execArgs = ['analyze', '-f', 'json'].concat(
+          getEngineArgs(linterEnabledEngines),
+          [relpath, '</dev/null']);
+        const execOpts = {
+          cwd: dirname(configurationFilePath),
+        };
+        if (this.disableTimeout) {
+          execOpts.timeout = Infinity;
+        }
 
         // Debug the command executed to run the Code Climate CLI to the console
         if (atom.inDevMode()) {
           // eslint-disable-next-line no-console
-          console.log(`linter-codeclimate:: Command: \`${cmd}\``);
+          console.log('linter-codeclimate:: Command: ' +
+            `\`${this.executablePath} ${execArgs.join(' ')}\``);
         }
 
         // Start measure for how long the analysis took.
         startMeasure('linter-codeclimate: Analysis');
 
         // Execute the Code Climate CLI, parse the results, and emit them to the
         // Linter package as warnings. The Linter package handles the styling.
-        const execArgs = ['-lc', cmd];
-        const execOpts = {
-          cwd: dirname(configurationFilePath),
-        };
-        const result = await Helpers.exec('/bin/bash', execArgs, execOpts);
+        let result;
+        try {
+          result = await Helpers.exec(this.executablePath, execArgs, execOpts);
+        } catch (e) {
+          notifyError(e, `${this.executablePath} ${execArgs.join(' ')}`);
+          return null;
+        }
         let messages;
         try {
           messages = JSON.parse(result);
         } catch (e) {
-          // eslint-disable-next-line no-console
-          console.error(
-            'Invalid JSON returned from codeclimate, error and output attached.',
-            e, result);
+          notifyError(e, `${this.executablePath} ${execArgs.join(' ')}`,
+            'Invalid JSON returned from CodeClimate.', `- RETURNED JSON: \`${result}\``);
```

Comment:
Since the returned JSON can be quite large (as your example screenshot shows), let's just log it to the console here, and change the message to "See the Console for details".